test/SILOptimizer/generic_specialization_loops_detection_with_loops.swift

```diff
@@ -0,0 +1,105 @@
+// RUN: %target-swift-frontend -O -emit-sil -Xllvm -sil-print-generic-specialization-loops -Xllvm -sil-print-generic-specialization-info %s 2>&1 | %FileCheck --check-prefix=CHECK %s
+
```

Comment:
I'm not sure about what you mean here. The functions are mutually recursive, which means that even with `@inline(__always)` the performance inliner woulds stop recursive inlining.

But coming back to your concern regarding the inlining actually happening. There is a line like this in the test:
`Check that the compiler has produced a specialization information for a call-site.` It checks for a specialization information on a call-site, which may only happen if a specialized function was inlined. Is it what you'd like to check? Or do you mean something else?